src/Agent.Plugins/TestFilePublisher/Plugin/TestFilePublisherLogPlugin.cs

```diff
@@ -0,0 +1,208 @@
+            try
+            {
+                _logger = _logger ?? new TraceLogger(context);
+                _telemetry = _telemetry ?? new TelemetryDataCollector(new ClientFactory(context.VssConnection), _logger);
+
+                await PopulatePipelineConfig(context);
+
+                _telemetry.AddOrUpdate(TelemetryConstants.PluginInitialized, true);
+                _telemetry.AddOrUpdate(TelemetryConstants.PluginDisabled, true);
+
+                if (DisablePlugin(context))
+                {
+                    return false; // disable the plugin
+                }
+
+                _testFilePublisher = _testFilePublisher ??
+                                     new Plugins.TestFilePublisher.TestFilePublisher(context.VssConnection, _pipelineConfig, new TestFileTraceListener(context));
+            }
+            catch (Exception ex)
+            {
+                context.Trace(ex.ToString());
+                _logger?.Warning($"Unable to initialize {FriendlyName}.");
+                _telemetry?.AddOrUpdate(TelemetryConstants.InitializeFailed, ex);
+                return false;
+            }
+            finally
+            {
+                if (_telemetry != null)
+                {
+                    await _telemetry.PublishCumulativeTelemetryAsync();
+                }
+            }
+
+            return true;
+        }
+
+        /// <inheritdoc />
+        public async Task ProcessLineAsync(IAgentLogPluginContext context, Pipelines.TaskStepDefinitionReference step, string line)
+        {
+            await Task.CompletedTask;
+        }
+
+        /// <inheritdoc />
+        public async Task FinalizeAsync(IAgentLogPluginContext context)
+        {
+            using (var timer = new SimpleTimer("Finalize", _logger, TimeSpan.FromMilliseconds(Int32.MaxValue),
+                new TelemetryDataWrapper(_telemetry, TelemetryConstants.FinalizeAsync)))
+            {
+                await _testFilePublisher.InitializeAsync();
+                await _testFilePublisher.PublishAsync();
+            }
+
+            await _telemetry.PublishCumulativeTelemetryAsync();
+        }
+
+        /// <summary>
+        /// Return true if plugin needs to be disabled
+        /// </summary>
+        private bool DisablePlugin(IAgentLogPluginContext context)
+        {
+            // do we want to log that the plugin is disabled due to x reason here?
+            if (context.Variables.TryGetValue("Agent.ForceEnable.TestFilePublisherLogPlugin", out var forceEnable)
+                && string.Equals("true", forceEnable.Value, StringComparison.OrdinalIgnoreCase))
+            {
+                return false;
+            }
+
+            // Enable only for build
+            if (!context.Variables.TryGetValue("system.hosttype", out var hostType)
+                || !string.Equals("Build", hostType.Value, StringComparison.OrdinalIgnoreCase))
+            {
+                _telemetry.AddOrUpdate("PluginDisabledReason", "NotABuild");
```

Comment:
>NotABuild [](start = 64, length = 9)

hosttype

Reply:
can you push the host type also to telemetry pls